surfactant/infoextractors/pe_file.py

```diff
@@ -242,9 +243,9 @@ def get_assemblyref_info(asmref_info):
     asmref: Dict[str, Any] = {}
     add_core_assembly_info(asmref, asmref_info)
     asmref["HashValue"] = (
-        asmref_info.HashValue.hex()
+        asmref_info.HashValue.hex().raw_data
         if hasattr(asmref_info.HashValue, "hex")
```

Comment:
This check should probably look to see if HashValue.raw_data has a "hex" attribute, and then the line above it call the hex function on raw_data.

I haven't taken a close look at the custom data types that dnfile has switched to, but depending on what it has maybe now it can be guaranteed that HashValue.raw_data will always exist and never be None (e.g. always have a hex() method).

Reply:
It looks like it'll always have the raw_data attribute; each heap stream returns a HeapItem, and, looking at the source, it seems like it'll always have the raw_data attribute (see [here](https://github.com/malwarefrank/dnfile/blob/fcccdafb00c37b4e47c038b4dd855408c69f2ece/src/dnfile/base.py#L120))

I'll go ahead and remove the if/else ternary

Reply:
Thanks for the HeapItem ref -- I changed it to `value` after reading up a bit more on `value` vs `raw_data` for the HeapItemBinary derived type in dnfile, and the use of compressed integers to represent string lengths in .NET (ECMA-335 spec section II.23.2).

Essentially `value` doesn't include the encoded length byte, which we don't need to preserve.